### ISSUE TITLE
Unable to find a suitable algorithm for doing forward convolution

### ISSUE THREAD

**Issue by tu6ge** [ROCmSoftwarePlatform/MIOpen#87]
```
MIOpen Error: /root/driver/streamhsa/MLOpen/src/tmp_dir.cpp:18: Can't execute cd /tmp/miopen-MIOpenUtilKernels.cl-75ac-3801-2f92-7633; /opt/rocm/bin/clang-ocl  -DNUM_CH_PER_WG=1 -DNUM_IM_BLKS_X=8 -DNUM_IM_BLKS=256 -DLOCAL_MEM_SIZE=532 -DSTRIDE_GT_1=0 -DTILE_SZ_X=32 -DTILE_SZ_Y=8 -DUSE_IM_OFF_GUARD=1 -DMIOPEN_USE_FP32=1 -mcpu=gfx803 -Wno-everything MIOpenUtilKernels.cl -o /tmp/miopen-MIOpenUtilKernels.cl-75ac-3801-2f92-7633/MIOpenUtilKernels.cl.o
2019-03-13 21:15:21.118018: F tensorflow/stream_executor/rocm/rocm_dnn.cc:2659] Check failed: status == miopenStatusSuccess (7 vs. 0)Unable to find a suitable algorithm for doing forward convolution
```
I set Environment is this
```
export MIOPEN_ENABLE_LOGGING=1
export MIOPEN_DEBUG_AMD_ROCM_PRECOMPILED_BINARIES=0
export MIOPEN_DEBUG_GCN_ASM_KERNELS=0
export MIOPEN_DEBUG_CONV_FFT=0
export MIOPEN_DEBUG_CONV_DIRECT=0
```
show info
```
MIOpen(HIP): miopenStatus_t miopenCreateTensorDescriptor(miopenTensorDescriptor_t *){
tensorDesc = 0
}
MIOpen(HIP): miopenStatus_t miopenSet4dTensorDescriptor(miopenTensorDescriptor_t, miopenDataType_t, int, int, int, int){
tensorDesc = 
dataType = 1
n = 16
c = 3
h = 262
w = 262
}
MIOpen(HIP): miopenStatus_t miopenCreateTensorDescriptor(miopenTensorDescriptor_t *){
tensorDesc = 0
}
MIOpen(HIP): miopenStatus_t miopenSet4dTensorDescriptor(miopenTensorDescriptor_t, miopenDataType_t, int, int, int, int){
tensorDesc = 
dataType = 1
n = 16
c = 64
h = 256
w = 256
}
MIOpen(HIP): miopenStatus_t miopenCreateTensorDescriptor(miopenTensorDescriptor_t *){
tensorDesc = 0
}
MIOpen(HIP): miopenStatus_t miopenSet4dTensorDescriptor(miopenTensorDescriptor_t, miopenDataType_t, int, int, int, int){
tensorDesc = 
dataType = 1
n = 64
c = 3
h = 7
w = 7
}
MIOpen(HIP): miopenStatus_t miopenCreateConvolutionDescriptor(miopenConvolutionDescriptor_t *){
convDesc = 0
}
MIOpen(HIP): miopenStatus_t miopenInitConvolutionDescriptor(miopenConvolutionDescriptor_t, miopenConvolutionMode_t, int, int, int, int, int, int){
convDesc = miopenConvolution, 0, 0, 1, 1, 1, 1, 
c_mode = 0
pad_h = 0
pad_w = 0
u = 1
v = 1
dilation_h = 1
dilation_w = 1
}
MIOpen(HIP): miopenStatus_t miopenSetConvolutionGroupCount(miopenConvolutionDescriptor_t, int){
convDesc = miopenConvolution, 0, 0, 1, 1, 1, 1, 
groupCount = 1
}
MIOpen(HIP): miopenStatus_t miopenConvolutionForwardGetWorkSpaceSize(miopenHandle_t, const miopenTensorDescriptor_t, const miopenTensorDescriptor_t, const miopenConvolutionDescriptor_t, const miopenTensorDescriptor_t, size_t *){
wDesc = 64, 3, 7, 7
yDesc = 16, 64, 256, 256
convDesc = miopenConvolution, 0, 0, 1, 1, 1, 1, 
workSpaceSize = 139901858015520
}
MIOpen(HIP): miopenStatus_t miopenFindConvolutionForwardAlgorithm(miopenHandle_t, const miopenTensorDescriptor_t, const void *, const miopenTensorDescriptor_t, const void *, const miopenConvolutionDescriptor_t, const miopenTensorDescriptor_t, void *, const int, int *, miopenConvAlgoPerf_t *, void *, size_t, bool){
xDesc = 16, 3, 262, 262
x = 0x50ae36c00
wDesc = 64, 3, 7, 7
w = 0x50a1a5100
convDesc = miopenConvolution, 0, 0, 1, 1, 1, 1, 
yDesc = 16, 64, 256, 256
y = 0x50bac8700
requestAlgoCount = 1
returnedAlgoCount = 0
perfResults = 
workSpace = 0x51bac8700
workSpaceSize = 38535168
exhaustiveSearch = 0
}
MIOpen(HIP): void miopen::DirConvFindCore(miopen::Handle &, const miopen::TensorDescriptor &, ConstData_t, const miopen::TensorDescriptor &, ConstData_t, const miopen::TensorDescriptor &, Data_t, size_t, const miopen::ConvolutionDescriptor &, bool, miopen::DbRecord &){
"convolution, non 1x1" = convolution, non 1x1
}
MIOpen Error: /root/driver/streamhsa/MLOpen/src/tmp_dir.cpp:18: Can't execute cd /tmp/miopen-MIOpenUtilKernels.cl-cad8-5324-1eae-c9e9; /opt/rocm/bin/clang-ocl  -DNUM_CH_PER_WG=1 -DNUM_IM_BLKS_X=8 -DNUM_IM_BLKS=256 -DLOCAL_MEM_SIZE=532 -DSTRIDE_GT_1=0 -DTILE_SZ_X=32 -DTILE_SZ_Y=8 -DUSE_IM_OFF_GUARD=1 -DMIOPEN_USE_FP32=1 -mcpu=gfx803 -Wno-everything MIOpenUtilKernels.cl -o /tmp/miopen-MIOpenUtilKernels.cl-cad8-5324-1eae-c9e9/MIOpenUtilKernels.cl.o
2019-03-13 21:21:59.181511: F tensorflow/stream_executor/rocm/rocm_dnn.cc:2659] Check failed: status == miopenStatusSuccess (7 vs. 0)Unable to find a suitable algorithm for doing forward convolution

```

**Comment by tu6ge** [comment]
I upgrade miopen-hip and try again,show info:
```
MIOpen Error: /root/driver/MLOpen/src/ocl/gcn_asm_utils.cpp:140: Error: X-AMDGCN-ASM: popen()
2019-03-13 21:45:52.245867: F tensorflow/stream_executor/rocm/rocm_dnn.cc:2659] Check failed: status == miopenStatusSuccess (7 vs. 0)Unable to find a suitable algorithm for doing forward convolution
```

**Comment by dagamayank** [comment]
This path looks extremely suspicious - `/root/driver/streamhsa`. @tu6ge what is your hw (GPU) / sw (ROCm version) configuration and how did you install MIOpen?

**Comment by pfultz2** [comment]
It seems to be unable to run `clang-ocl`. Do you have it installed?

**Comment by tu6ge** [comment]
use HIP

**Comment by atamazov** [comment]
@tu6ge This may sound strange, but you need `clang-ocl` to build and run MIOpen with HIP backend.

**Comment by tu6ge** [comment]
build clang-ocl is error
```
Could NOT find ROCM (missing: ROCM_DIR)
CMake Error at CMakeLists.txt:9 (include):
  include could not find load file:

    ROCMSetupVersion


CMake Error at CMakeLists.txt:10 (include):
  include could not find load file:

    ROCMCreatePackage


CMake Error at CMakeLists.txt:13 (rocm_setup_version):
  Unknown CMake command "rocm_setup_version".


-- Configuring incomplete, errors occurred!
```

**Comment by tu6ge** [comment]
clang-ocl is installed

**Comment by tu6ge** [comment]
new return error info:
```
MIOpen Error: /root/driver/MLOpen/src/ocl/gcn_asm_utils.cpp:140: Error: X-AMDGCN-ASM: popen()
2019-03-14 21:16:24.406450: F tensorflow/stream_executor/rocm/rocm_dnn.cc:2659] Check failed: status == miopenStatusSuccess (7 vs. 0)Unable to find a suitable algorithm for doing forward convolution
```

**Comment by atamazov** [comment]
Do you still have these settings in env?
```
export MIOPEN_DEBUG_AMD_ROCM_PRECOMPILED_BINARIES=0
export MIOPEN_DEBUG_GCN_ASM_KERNELS=0
export MIOPEN_DEBUG_CONV_FFT=0
export MIOPEN_DEBUG_CONV_DIRECT=0
```

**Comment by tu6ge** [comment]
my rocm is not running,I need to fix this first.

**Comment by ali-masoudi** [comment]
I got this error too
```
MIOpen Error: /root/driver/MLOpen/src/tmp_dir.cpp:18: Can't execute cd /tmp/miopen-tinygemm.cl-62e7-9c82-8667-8256; /opt/rocm/bin/clang-ocl  -mcpu=gfx803 -Wno-everything tinygemm.cl -o /tmp/miopen-tinygemm.cl-62e7-9c82-8667-8256/tinygemm.cl.o
2019-04-02 03:20:42.459606: F tensorflow/stream_executor/rocm/rocm_dnn.cc:2659] Check failed: status == miopenStatusSuccess (7 vs. 0)Unable to find a suitable algorithm for doing forward convolution
Aborted
```

Ubuntu 18.04 Kernel 4.18.0-16
```
/opt/rocm/bin/clang-ocl --version
clang version 8.0 
Target: amdgcn-amd-amdhsa-amdgizcl
Thread model: posix
InstalledDir: /opt/rocm/opencl/bin/x86_64
/opt/rocm/opencl/bin/x86_64/llvm-link: .orig.bc: error: Could not open input file: No such file or directory
/opt/rocm/opencl/bin/x86_64/llvm-link: error:  loading file '.orig.bc'
clang version 8.0 
Target: amdgcn-amd-amdhsa-amdgizcl
Thread model: posix
InstalledDir: /opt/rocm/opencl/bin/x86_64
```

**Comment by JohnBFrm** [comment]
I seem to be hitting the same error message. Attempting to run https://github.com/danielegrattarola/twitter-sentiment-cnn with my own dataset. Is it perhaps an OOM error?

`MIOpen Error: /root/driver/MLOpen/src/tmp_dir.cpp:18: Can't execute cd /tmp/miopen-MIOpenIm2d2Col.cl-c204-47b2-63c1-3119; /opt/rocm/bin/clang-ocl  -DNUM_CH_PER_WG=1 -DNUM_IM_BLKS_X=1 -DNUM_IM_BLKS=459 -DLOCAL_MEM_SIZE=1590 -DSTRIDE_GT_1=0 -DTILE_SZ_X=32 -DTILE_SZ_Y=8 -DUSE_IM_OFF_GUARD=1 -DMIOPEN_USE_FP32=1 -mcpu=gfx906 -Wno-everything MIOpenIm2d2Col.cl -o /tmp/miopen-MIOpenIm2d2Col.cl-c204-47b2-63c1-3119/MIOpenIm2d2Col.cl.o
2019-04-14 19:28:24.865848: F tensorflow/stream_executor/rocm/rocm_dnn.cc:2659] Check failed: status == miopenStatusSuccess (7 vs. 0)Unable to find a suitable algorithm for doing forward convolution
Aborted (core dumped)
`

**Comment by atamazov** [comment]
@JohnBFrm We need additional info. Could you please do the following:
- Set in you envurinment (see [here](https://github.com/AMDComputeLibraries/MLOpen/blob/develop/doc/src/DebugAndLogging.md) for more info):
```
export MIOPEN_ENABLE_LOGGING_CMD=1
export MIOPEN_ENABLE_LOGGING=1
export MIOPEN_LOG_LEVEL=5
```
- Run your application with this environment and collect log from console (`... 2>&1 | tee logfile.txt`)
- Attach resulting `logfile.txt` to this ticket.

Thanks.

**Comment by JohnBFrm** [comment]
> @JohnBFrm We need additional info. Could you please do the following:
> 
> * Set in you envurinment (see [here](https://github.com/AMDComputeLibraries/MLOpen/blob/develop/doc/src/DebugAndLogging.md) for more info):
> 
> ```
> export MIOPEN_ENABLE_LOGGING_CMD=1
> export MIOPEN_ENABLE_LOGGING=1
> export MIOPEN_LOG_LEVEL=5
> ```
> * Run your application with this environment and collect log from console (`... 2>&1 | tee logfile.txt`)
> * Attach resulting `logfile.txt` to this ticket.
> 
> Thanks.

Hi Artem,

Done! Attached
[logfile.txt](https://github.com/ROCmSoftwarePlatform/MIOpen/files/3080520/logfile.txt)

**Comment by atamazov** [comment]
@JohnBFrm It works for me even on gfx900.
There is Winograd kernel which is the fastest.

Do you still have these settings in env?
```
export MIOPEN_DEBUG_AMD_ROCM_PRECOMPILED_BINARIES=0
export MIOPEN_DEBUG_GCN_ASM_KERNELS=0
export MIOPEN_DEBUG_CONV_FFT=0
export MIOPEN_DEBUG_CONV_DIRECT=0
```

**Comment by atamazov** [comment]
2all: driver command to reproduce:
```
./bin/MIOpenDriver conv -n 128 -c 1 -H 3669 -W 128 -k 128 -c 1 -y 3 -x 128 -p 0 -q 0 -V 0 -t 1 -w 1 -F 1 -i 1
```

**Comment by atamazov** [comment]
@JohnBFrm Looks like you have a problem with GEMM Im2Col path. Please attach log of this:
```
./bin/MIOpenDriver conv -c 32 -V 0 -t 1 -w 1 -F 1 -i 1
```
Please set only `export MIOPEN_LOG_LEVEL=5` and unset all other MIOPEN envvars. Thanks.

**Comment by JohnBFrm** [comment]
@atamazov Done, output isn't large so I'm adding it as code:
`comp@andromeda:~/MIOpen/build$ ./bin/MIOpenDriver conv -c 32 -V 0 -t 1 -w 1 -F 1 -i 1
MIOpenDriver: conv -c 32 -V 0 -t 1 -w 1 -F 1 -i 1
MIOpen(HIP): Info [DetectAmdRocmMetadataVersion] AMDHSA_1_0
MIOpen(HIP): Info [FindSolutionImpl] ConvOclDirectFwd
MIOpen(HIP): Info [FindRecordUnsafe] File is unreadable: /home/john/.config/miopen//gfx906_60.cd.updb.txt
MIOpen(HIP): Info [FindSolutionImpl] Perf Db: record not found for: ConvOclDirectFwd
MIOpen(HIP): Info [SetValues] 32-32-32-3x3-32-30-30-100-0x0-1x1-1x1-0-NCHW-FP32-F, content inserted: miopenConvolutionFwdAlgoGEMM:gemm,3.024,1036800,0_0_900_288_900_900_32_288
MIOpen(HIP): Info [FindSolutionImpl] ConvBinWinogradRxS (not searchable)
MIOpen(HIP): Info [SetValues] 32-32-32-3x3-32-30-30-100-0x0-1x1-1x1-0-NCHW-FP32-F, content inserted: miopenConvolutionFwdAlgoWinograd:ConvBinWinogradRxS,0.12496,0,32x32x32x3x3x32x30x30x100xNCHWxFP32x0x0x1x1x1x1x1x1
MIOpen(HIP): Info [FindSolutionImpl] ConvOclDirectFwd
MIOpen(HIP): Info [FindRecordUnsafe] File is unreadable: /home/john/.config/miopen//gfx906_60.cd.updb.txt
MIOpen(HIP): Info [FindSolutionImpl] Perf Db: record not found for: ConvOclDirectFwd
MIOpen(HIP): Info [DirConvFindCore] ConvOclDirectFwd: MIOpenConvUni: 0.455998 < 3.40282e+38
MIOpen(HIP): Info [DirConvFindCore] Selected: ConvOclDirectFwd: MIOpenConvUni: 0.455998, workspce_sz = 0
MIOpen(HIP): Info [SetValues] 32-32-32-3x3-32-30-30-100-0x0-1x1-1x1-0-NCHW-FP32-F, content inserted: miopenConvolutionFwdAlgoDirect:ConvOclDirectFwd,0.455998,0,32x32x32x3x3x32x30x30x100xNCHWxFP32x0x0x1x1x1x1x1x1
MIOpen(HIP): Info [FindConvFwdAlgorithm] miopenConvolutionFwdAlgoWinograd	0.12496	0
MIOpen(HIP): Info [FindConvFwdAlgorithm] miopenConvolutionFwdAlgoDirect	0.455998	0
MIOpen(HIP): Info [FindConvFwdAlgorithm] miopenConvolutionFwdAlgoGEMM	3.024	1036800
MIOpen(HIP): Info [FindConvFwdAlgorithm] FW Chosen Algorithm: ConvBinWinogradRxS , 0, 0.12496
Wall-clock Time Forward Conv. Elapsed: 0.171923 ms
MIOpen Forward Conv. Algorithm: 3
GPU Kernel Time Forward Conv. Elapsed: 0.124480 ms (average)
stats: name, n, c, ho, wo, x, y, k, flopCnt, bytesRead, bytesWritten, GFLOPs, GB/s, timeMs
stats: fwd-conv3x3u1, 100, 32, 30, 30, 3, 3, 32,  1658880000, 13144064, 11520000, 13326, 198, 0.124480`

**Comment by JohnBFrm** [comment]
Had trouble with the post, readding it:

@atamazov Done, output isn't large so I'm adding it as a quote:

> comp@andromeda:~/MIOpen/build$ ./bin/MIOpenDriver conv -c 32 -V 0 -t 1 -w 1 -F 1 -i 1
> MIOpenDriver: conv -c 32 -V 0 -t 1 -w 1 -F 1 -i 1
> MIOpen(HIP): Info [DetectAmdRocmMetadataVersion] AMDHSA_1_0
> MIOpen(HIP): Info [FindSolutionImpl] ConvOclDirectFwd
> MIOpen(HIP): Info [FindRecordUnsafe] File is unreadable: /home/comp/.config/miopen//gfx906_60.cd.updb.txt
> MIOpen(HIP): Info [FindSolutionImpl] Perf Db: record not found for: ConvOclDirectFwd
> MIOpen(HIP): Info [SetValues] 32-32-32-3x3-32-30-30-100-0x0-1x1-1x1-0-NCHW-FP32-F, content inserted: miopenConvolutionFwdAlgoGEMM:gemm,3.024,1036800,0_0_900_288_900_900_32_288
> MIOpen(HIP): Info [FindSolutionImpl] ConvBinWinogradRxS (not searchable)
> MIOpen(HIP): Info [SetValues] 32-32-32-3x3-32-30-30-100-0x0-1x1-1x1-0-NCHW-FP32-F, content inserted: miopenConvolutionFwdAlgoWinograd:ConvBinWinogradRxS,0.12496,0,32x32x32x3x3x32x30x30x100xNCHWxFP32x0x0x1x1x1x1x1x1
> MIOpen(HIP): Info [FindSolutionImpl] ConvOclDirectFwd
> MIOpen(HIP): Info [FindRecordUnsafe] File is unreadable: /home/comp/.config/miopen//gfx906_60.cd.updb.txt
> MIOpen(HIP): Info [FindSolutionImpl] Perf Db: record not found for: ConvOclDirectFwd
> MIOpen(HIP): Info [DirConvFindCore] ConvOclDirectFwd: MIOpenConvUni: 0.455998 < 3.40282e+38
> MIOpen(HIP): Info [DirConvFindCore] Selected: ConvOclDirectFwd: MIOpenConvUni: 0.455998, workspce_sz = 0
> MIOpen(HIP): Info [SetValues] 32-32-32-3x3-32-30-30-100-0x0-1x1-1x1-0-NCHW-FP32-F, content inserted: miopenConvolutionFwdAlgoDirect:ConvOclDirectFwd,0.455998,0,32x32x32x3x3x32x30x30x100xNCHWxFP32x0x0x1x1x1x1x1x1
> MIOpen(HIP): Info [FindConvFwdAlgorithm] miopenConvolutionFwdAlgoWinograd	0.12496	0
> MIOpen(HIP): Info [FindConvFwdAlgorithm] miopenConvolutionFwdAlgoDirect	0.455998	0
> MIOpen(HIP): Info [FindConvFwdAlgorithm] miopenConvolutionFwdAlgoGEMM	3.024	1036800
> MIOpen(HIP): Info [FindConvFwdAlgorithm] FW Chosen Algorithm: ConvBinWinogradRxS , 0, 0.12496
> Wall-clock Time Forward Conv. Elapsed: 0.171923 ms
> MIOpen Forward Conv. Algorithm: 3
> GPU Kernel Time Forward Conv. Elapsed: 0.124480 ms (average)
> stats: name, n, c, ho, wo, x, y, k, flopCnt, bytesRead, bytesWritten, GFLOPs, GB/s, timeMs
> stats: fwd-conv3x3u1, 100, 32, 30, 30, 3, 3, 32,  1658880000, 13144064, 11520000, 13326, 198, 0.124480

**Comment by atamazov** [comment]
@JohnBFrm Please double check if proper versions of ROCm, rocBLAS etc are installed. [The following driver command](https://github.com/ROCmSoftwarePlatform/MIOpen/issues/87#issuecomment-483283137) works for me. I've tried with MIOpen1.8 release, using ROCm 2.3, rocBLAS v2.2:
```
$ dpkg -l | grep -i -E "rocm|rocblas"
ii  rocblas                                    2.2.2.0                                      amd64        Radeon Open Compute BLAS library
ii  rocm-clang-ocl                             0.4.0-7ce124f                                amd64        OpenCL compilation with clang compiler.
ii  rocm-dev                                   2.3.14                                       amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-dkms                                  2.3.14                                       amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-opencl                                1.2.0-2019040843                             amd64        OpenCL/ROCm
ii  rocm-opencl-dev                            1.2.0-2019040843                             amd64        OpenCL/ROCm
ii  rocm-profiler                              5.6.7259                                     amd64        The ROCm-Profiler is a command-line profiler for profiling applications running on the Radeon Open Compute platforms.
ii  rocm-utils                                 2.3.14                                       amd64        Radeon Open Compute (ROCm) Runtime software stack
...

$ env | grep -E "MIO|AMD" | sort
MIOPEN_LOG_LEVEL=5

$ ./bin/MIOpenDriver conv -n 128 -c 1 -H 3669 -W 128 -k 128 -c 1 -y 3 -x 128 -p 0 -q 0 -V 0 -t 1 -w 1 -F 1 -i 1
MIOpenDriver: conv -n 128 -c 1 -H 3669 -W 128 -k 128 -c 1 -y 3 -x 128 -p 0 -q 0 -V 0 -t 1 -w 1 -F 1 -i 1
MIOpen(HIP): Info [DetectAmdRocmMetadataVersion] AMDHSA_1_0
MIOpen(HIP): Info [SetValues] 1-3669-128-3x128-128-3667-1-128-0x0-1x1-1x1-0-NCHW-FP32-F, content inserted: miopenConvolutionFwdAlgoGEMM:gemm,33.3581,5632512,0_0_3667_384_3667_3667_128_384
MIOpen(HIP): Info [FindSolutionImpl] ConvBinWinogradRxS (not searchable)
MIOpen(HIP): Info [SetValues] 1-3669-128-3x128-128-3667-1-128-0x0-1x1-1x1-0-NCHW-FP32-F, content inserted: miopenConvolutionFwdAlgoWinograd:ConvBinWinogradRxS,17.0736,0,1x3669x128x3x128x128x3667x1x128xNCHWxFP32x0x0x1x1x1x1x1x1
MIOpen(HIP): Info [FindConvFwdAlgorithm] miopenConvolutionFwdAlgoWinograd       17.0736 0
MIOpen(HIP): Info [FindConvFwdAlgorithm] miopenConvolutionFwdAlgoGEMM   33.3581 5632512
MIOpen(HIP): Info [FindConvFwdAlgorithm] FW Chosen Algorithm: ConvBinWinogradRxS , 0, 17.0736
Wall-clock Time Forward Conv. Elapsed: 17.210855 ms
MIOpen Forward Conv. Algorithm: 3
GPU Kernel Time Forward Conv. Elapsed: 17.098965 ms (average)
stats: name, n, c, ho, wo, x, y, k, flopCnt, bytesRead, bytesWritten, GFLOPs, GB/s, timeMs
stats: fwd-conv3x128u1, 128, 1, 3667, 1, 3, 128, 128,  46141538304, 240648192, 240320512, 2698, 28, 17.098965
```

**Comment by daniellowell** [comment]
@JohnBFrm Any update on this ticket?

**Comment by JohnBFrm** [comment]
@daniellowell @atamazov Still having the same issue. Posting results of requested commands:
`dpkg -l | grep -i -E "rocm|rocblas"
ii  hsa-ext-rocr-dev                           1.1.9-68-gc862c1c                            amd64        AMD Heterogeneous System Architecture HSA - Linux HSA Runtime extensions for ROCm platforms
ii  hsa-rocr-dev                               1.1.9-68-gc862c1c                            amd64        AMD Heterogeneous System Architecture HSA - Linux HSA Runtime for ROCm platforms
ii  rocblas                                    2.2.5.0                                      amd64        Radeon Open Compute BLAS library
ii  rocm-clang-ocl                             0.4.0-7ce124f                                amd64        OpenCL compilation with clang compiler.
ii  rocm-dev                                   2.4.25                                       amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-device-libs                           0.0.1                                        amd64        Radeon Open Compute - device libraries
ii  rocm-dkms                                  2.4.25                                       amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-libs                                  2.4.25                                       amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-opencl                                1.2.0-2019050231                             amd64        OpenCL/ROCm
ii  rocm-opencl-dev                            1.2.0-2019050231                             amd64        OpenCL/ROCm
ii  rocm-profiler                              5.6.7262                                     amd64        The ROCm-Profiler is a command-line profiler for profiling applications running on the Radeon Open Compute platforms.
ii  rocm-smi                                   1.0.0-129-g71924de                           amd64        System Management Interface for ROCm
ii  rocm-utils                                 2.4.25                                       amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm_bandwidth_test                        1.0.0                                        amd64        Test to measure PciE bandwidth on ROCm platforms
ii  rocminfo                                   1.0.0                                        amd64        Radeon Open Compute (ROCm) Runtime rocminfo tool
ii  rocr_debug_agent                           1.0.0                                        amd64        Radeon Open Compute (ROCm) Runtime debug agent`

`env | grep -E "MIO|AMD" | sort
MIOPEN_LOG_LEVEL=5
`

`./bin/MIOpenDriver conv -n 128 -c 1 -H 3669 -W 128 -k 128 -c 1 -y 3 -x 128 -p 0 -q 0 -V 0 -t 1 -w 1 -F 1 -i 1
MIOpenDriver: conv -n 128 -c 1 -H 3669 -W 128 -k 128 -c 1 -y 3 -x 128 -p 0 -q 0 -V 0 -t 1 -w 1 -F 1 -i 1
MIOpen(HIP): Info [DetectAmdRocmMetadataVersion] AMDHSA_1_0
MIOpen(HIP): Info [SetValues] 1-3669-128-3x128-128-3667-1-128-0x0-1x1-1x1-0-NCHW-FP32-F, content inserted: miopenConvolutionFwdAlgoGEMM:gemm,16.7322,5632512,0_0_3667_384_3667_3667_128_384
MIOpen(HIP): Info [FindSolutionImpl] ConvBinWinogradRxS (not searchable)
MIOpen(HIP): Info [SetValues] 1-3669-128-3x128-128-3667-1-128-0x0-1x1-1x1-0-NCHW-FP32-F, content inserted: miopenConvolutionFwdAlgoWinograd:ConvBinWinogradRxS,9.70126,0,1x3669x128x3x128x128x3667x1x128xNCHWxFP32x0x0x1x1x1x1x1x1
MIOpen(HIP): Info [FindConvFwdAlgorithm] miopenConvolutionFwdAlgoWinograd	9.70126	0
MIOpen(HIP): Info [FindConvFwdAlgorithm] miopenConvolutionFwdAlgoGEMM	16.7322	5632512
MIOpen(HIP): Info [FindConvFwdAlgorithm] FW Chosen Algorithm: ConvBinWinogradRxS , 0, 9.70126
Wall-clock Time Forward Conv. Elapsed: 8.641413 ms
MIOpen Forward Conv. Algorithm: 3
GPU Kernel Time Forward Conv. Elapsed: 8.596301 ms (average)
stats: name, n, c, ho, wo, x, y, k, flopCnt, bytesRead, bytesWritten, GFLOPs, GB/s, timeMs
stats: fwd-conv3x128u1, 128, 1, 3667, 1, 3, 128, 128,  46141538304, 240648192, 240320512, 5368, 56, 8.596301`

**Comment by atamazov** [comment]
How much system RAM do you have installed? If you run multiple miopen-enabled processes or threads in parallel, the OpenCL compiler might fail due to exhaustion of virtual memory.

Could you please attach fresh log (like you did [here](https://github.com/ROCmSoftwarePlatform/MIOpen/issues/87#issuecomment-483266984)).

Please also tell us how important is this for you. Thanks.

**Comment by JohnBFrm** [comment]
@atamazov I have 256 GB of system RAM. The highest I've seen with this dataset is around 160 GB used.
I'm only attempting to run this single task, which runs fine if run entirely on the CPU.

I'd like to get this working, but at the very least I'd like to understand the issue so that I can attempt to work around it, or be aware of it moving forward. Right now I am unable to run my project. Is it perhaps entirely a memory issue? Insufficient VRAM?

Log is from a test environment, nothing is omitted

[logfile.txt](https://github.com/ROCmSoftwarePlatform/MIOpen/files/3251568/logfile.txt)

**Comment by atamazov** [comment]
I've tried the failing config with both MIOpen downloaded from the public repo and our current development tip (release candidate). Both are working fine.

From your log:
```
MIOpen(HIP): Info [GcnAssemblerHasBug34765Impl] Detected
MIOpen(HIP): Info [DetectAmdRocmMetadataVersion] V3
```
This means that you have (and/or MIOpen sees and uses) some very old components of ROCm. Metadata V3 support is already deprecated (V1, V2, V3 are dismissed and only AMDHSA_1_0 and newer will be supported after next release). Bug34765 have been fixed long time ago as well.

This is confusing because `dpkg` output shows that you have a bit newer rocm components than me, but MIOpen reports older metadata (which means older runtime). After closer look at `dpkg` output, I see that you do not have `rock-dkms` installed. This probably means that you are using upstream kernel driver instead of AMD's package, see https://github.com/RadeonOpenCompute/ROCm#rocm-support-in-upstream-linux-kernels for more info.

Please fix this and try again.

Please also check your MIOpenGemm installation. I have it installed from github like this:
```bash
cd ~/github
git clone https://github.com/ROCmSoftwarePlatform/MIOpenGEMM
mkdir MIOpenGEMM/build
cd ~/github/MIOpenGEMM/build
cmake ..
make miopengemm
sudo make install
```

```bash
$ git log -1 --oneline
9547fb9 (HEAD -> master, origin/master, origin/HEAD) Merge pull request #34 from ROCmSoftwarePlatform/distance_fix
```
___Note:___ You need to rebuild MIOpen from the scratch for the above to take effect.

## If the above does not help

Set logging level to 6:
```
export MIOPEN_ENABLE_LOGGING_CMD=1
export MIOPEN_ENABLE_LOGGING=1
export MIOPEN_LOG_LEVEL=6
```
then collect output of
```
./bin/MIOpenDriver conv -n 128 -c 1 -H 3669 -W 128 -k 128 -c 1 -y 3 -x 128 -p 0 -q 0 -V 0 -t 1 -w 1 -F 1 -i 1
```
and attach the resulting log.

Also please collect tensorflow output with the same logging level (6) and attach that log too.

**Comment by JohnBFrm** [comment]
@atamazov The command for the dkpg was only for rocm and rocblas. If I filter for rock, I get:
```ii  rock-dkms                                  2.4-25                                       all          rock-dkms driver in DKMS format.```
I made specifically sure that I install AMD's package and not the upstream kernel.

I'll look over the rest of your post, too!

**Comment by atamazov** [comment]
What is really strange is that in spite of you have ROCm 2.4 installed, MIOpen reports
```
MIOpen(HIP): Info [GcnAssemblerHasBug34765Impl] Detected
```
which have been fixed in ROCm 1.7.

**Comment by JohnBFrm** [comment]
@atamazov I have no problem doing a full reinstall of the environment so that we're not chasing ghosts. Is there a setup procedure you'd like for me to follow?

**Comment by atamazov** [comment]
Nothing special. I simply follow instructions from https://github.com/RadeonOpenCompute/ROCm.

**Comment by JohnBFrm** [comment]
@atamazov That's the one I've followed, chapter Ubuntu Support - installing from a Debian repository

**Comment by JohnBFrm** [comment]
I tried reinstalling without any success.

I then changed my dataset a bit. This particular setup loads two files consisting of positive and negative responses to teach the network sentiments, one response per line. Certain lines could be well over a thousand characters. I decided to split these up into multiple lines instead of no more than 80 characters. I have no issue running it now.

If interested, I can provide the files for further testing.

**Comment by thusinh1969** [comment]
My problem is exactly the same with Keras Densenet. Simply stuck at inference time.

 /root/driver/MLOpen/src/tmp_dir.cpp:18: Can't execute cd /tmp/miopen-MIOpenUtilKernels4.cl-a846-aa58-441f-b8bb; /opt/rocm/bin/clang-ocl  -DMIOPEN_USE_FP16=0 -DMIOPEN_USE_FP32=1 -DMIOPEN_USE_BFP16=0  -DMIOPEN_USE_INT8=0 -DMIOPEN_USE_INT8x4=0 -DNC_TRANS_NCHW_OPT=0 -DNC_TRANS_CNHW_OPT=1 -DNC_TRANS_NCHW=0 -DNC_TRANS_CNHW=0 -DIN_OFF=92512 -DOUT_OFF=0 -DH=7 -DW=7 -DN=1 -DC=128 -DRD_BLCK=1 -DHW_RD=49 -DMAP_RD=6272 -DREAD_TYPE=float -DIS_2D_WG=1 -mcpu=gfx906 -Wno-everything MIOpenUtilKernels4.cl -o /tmp/miopen-MIOpenUtilKernels4.cl-a846-aa58-441f-b8bb/MIOpenUtilKernels4.cl.o
2019-07-29 02:09:00.690580: F tensorflow/stream_executor/rocm/rocm_dnn.cc:2707] Check failed: status == miopenStatusSuccess (7 vs. 0)Unable to find a suitable algorithm for doing forward convolution

**Comment by patsch007** [comment]
I just hit the same Problem using the latest version of ROCM.
Is there already a solution available ?

[ErrorMessage.txt](https://github.com/ROCmSoftwarePlatform/MIOpen/files/4188938/ErrorMessage.txt)
[Versions.txt](https://github.com/ROCmSoftwarePlatform/MIOpen/files/4188939/Versions.txt)

**Comment by daniellowell** [comment]
@patsch007 Please try with the latest ROCm and MIOpen. Feel free to reopen this ticket if you need more help.

**Comment by tonysy** [comment]
@daniellowell 

Hi, I run the https://github.com/facebookresearch/detectron2 , which support to install on AMD GPU, and run the training command , the network seemed to stuck, and the log is here:
[train_detectron2_log.log](https://github.com/ROCmSoftwarePlatform/MIOpen/files/4813408/train_detectron2_log.log)


The detail for MIOpen can be found in logs.

Software version:
```
[32m[06/22 18:57:01 pluses]: [0mEnvironment info:
---------------------  --------------------------------------------------------------------------------------------------------------------
sys.platform           linux
Python                 3.6.8 (default, Aug 10 2019, 04:11:00) [GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]
numpy                  1.16.4
pluses                 0.1.0 @/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses
Compiler               GCC 7.3
CUDA compiler          HIP 20.12.6
DETECTRON2_ENV_MODULE  <not set>
PyTorch                1.5.0a0+4ff3872 @/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch
PyTorch debug build    False
GPU available          True
GPU 0,1,2,3            Device 66a1
ROCM_HOME              /public/software/compiler/rocm/rocm-3.3.0/
Pillow                 5.2.0
torchvision            0.6.0a0+abc1506 @/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torchvision
fvcore                 0.1.1.post20200622
---------------------  --------------------------------------------------------------------------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - HIP Runtime 20.12.6
  - MIOpen 2.3.0
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_INTERNAL_THREADPOOL_IMPL -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, FORCE_FALLBACK_CUDA_MPI=1, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=OFF, USE_EIGEN_FOR_BLAS=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=OFF, USE_MKLDNN=OFF, USE_MPI=1, USE_NCCL=1, USE_NNPACK=ON, USE_OPENMP=1, USE_STATIC_DISPATCH=OFF, 
```

When I press ctrl+c, I got this, hope you give some suggestiosn and help, thanks
```
MIOpen Error: /root/driver/MLOpen/src/tmp_dir.cpp:47: Can't execute cd /tmp/miopen-gridwise_convolution_implicit_gemm_v4r4_nchw_kcyx_nkhw.cpp-a13b-b2de-0966-1230; KMOPTLLC="-mattr=+enable-ds128 -amdgpu-enable-global-sgpr-addr --amdgpu-spill-vgpr-to-agpr=0" /opt/rocm-3.3.0/bin/hcc  -std=c++14  -DCK_PARAM_PROBLEM_N=4 -DCK_PARAM_PROBLEM_K=256 -DCK_PARAM_PROBLEM_C=64 -DCK_PARAM_PROBLEM_HI=336 -DCK_PARAM_PROBLEM_WI=200 -DCK_PARAM_PROBLEM_HO=336 -DCK_PARAM_PROBLEM_WO=200 -DCK_PARAM_PROBLEM_Y=1 -DCK_PARAM_PROBLEM_X=1 -DCK_PARAM_PROBLEM_CONV_STRIDE_H=1 -DCK_PARAM_PROBLEM_CONV_STRIDE_W=1 -DCK_PARAM_PROBLEM_CONV_DILATION_H=1 -DCK_PARAM_PROBLEM_CONV_DILATION_W=1 -DCK_PARAM_PROBLEM_IN_LEFT_PAD_H=0 -DCK_PARAM_PROBLEM_IN_LEFT_PAD_W=0 -DCK_PARAM_PROBLEM_IN_RIGHT_PAD_H=0 -DCK_PARAM_PROBLEM_IN_RIGHT_PAD_W=0 -DCK_PARAM_PROBLEM_CONV_DIRECTION_FORWARD=1 -DCK_PARAM_PROBLEM_CONV_DIRECTION_BACKWARD_DATA=0 -DCK_PARAM_PROBLEM_CONV_DIRECTION_BACKWARD_WEIGHT=0 -DCK_PARAM_TUNABLE_BLOCK_SIZE=256 -DCK_PARAM_TUNABLE_GEMM_M_PER_BLOCK=128 -DCK_PARAM_TUNABLE_GEMM_N_PER_BLOCK=128 -DCK_PARAM_TUNABLE_GEMM_K_PER_BLOCK=16 -DCK_PARAM_TUNABLE_GEMM_M_PER_THREAD=4 -DCK_PARAM_TUNABLE_GEMM_N_PER_THREAD=4 -DCK_PARAM_TUNABLE_GEMM_M_LEVEL0_CLUSTER=4 -DCK_PARAM_TUNABLE_GEMM_N_LEVEL0_CLUSTER=4 -DCK_PARAM_TUNABLE_GEMM_M_LEVEL1_CLUSTER=4 -DCK_PARAM_TUNABLE_GEMM_N_LEVEL1_CLUSTER=4 -DCK_PARAM_TUNABLE_GEMM_A_BLOCK_COPY_CLUSTER_LENGTHS_GEMM_K=4 -DCK_PARAM_TUNABLE_GEMM_A_BLOCK_COPY_CLUSTER_LENGTHS_GEMM_M=64 -DCK_PARAM_TUNABLE_GEMM_A_BLOCK_COPY_SRC_DATA_PER_READ_GEMM_K=4 -DCK_PARAM_TUNABLE_GEMM_A_BLOCK_COPY_DST_DATA_PER_WRITE_GEMM_M=2 -DCK_PARAM_TUNABLE_GEMM_B_BLOCK_COPY_CLUSTER_LENGTHS_GEMM_K=8 -DCK_PARAM_TUNABLE_GEMM_B_BLOCK_COPY_CLUSTER_LENGTHS_GEMM_N=32 -DCK_PARAM_TUNABLE_GEMM_B_BLOCK_COPY_SRC_DATA_PER_READ_GEMM_N=4 -DCK_PARAM_TUNABLE_GEMM_B_BLOCK_COPY_DST_DATA_PER_WRITE_GEMM_N=4 -DCK_PARAM_TUNABLE_GEMM_C_THREAD_COPY_DST_DATA_PER_WRITE_GEMM_N1=4 -DCK_PARAM_DEPENDENT_GRID_SIZE=4200 -DCK_THREADWISE_GEMM_USE_AMD_INLINE_ASM=1 -D__HIP_PLATFORM_HCC__=1 -DMIOPEN_USE_FP16=0 -DMIOPEN_USE_FP32=1 -DMIOPEN_USE_INT8=0 -DMIOPEN_USE_INT8x4=0 -DMIOPEN_USE_BFP16=0 -DMIOPEN_USE_INT32=0 -DMIOPEN_USE_RNE_BFLOAT16=1 -mcpu=gfx906 -Wno-everything -amdgpu-target=gfx906 -mcode-object-v3 -Wno-unused-command-line-argument -I. -isystem /opt/rocm-3.3.0/hip/../include -isystem /opt/rocm-3.3.0/hip/include -isystem /opt/rocm/include -hc -Wl,--enable-new-dtags -hc -L /opt/rocm/lib -Wl,-rpath /opt/rocm/lib -Wl,--whole-archive -hc -fPIC -std=c++14 -isystem /opt/rocm/include -isystem /opt/rocm/include -ldl -hc -fPIC -std=c++14 -isystem /opt/rocm/include -isystem /opt/rocm/include -Wl,--no-whole-archive -ldl -lm -hc -fPIC -std=c++14 -isystem /opt/rocm/include -isystem /opt/rocm/include gridwise_convolution_implicit_gemm_v4r4_nchw_kcyx_nkhw.cpp -o /tmp/miopen-gridwise_convolution_implicit_gemm_v4r4_nchw_kcyx_nkhw.cpp-a13b-b2de-0966-1230/gridwise_convolution_implicit_gemm_v4r4_nchw_kcyx_nkhw.cpp.o
MIOpen(HIP): miopenStatus_t miopenDestroyConvolutionDescriptor(miopenConvolutionDescriptor_t){
MIOpen(HIP): 	convDesc = conv2d, miopenConvolution, miopenPaddingDefault, {0, 0}, {1, 1}, {1, 1}, 
MIOpen(HIP): }
MIOpen(HIP): miopenStatus_t miopenDestroyTensorDescriptor(miopenTensorDescriptor_t){
MIOpen(HIP): 	tensorDesc = 256, 64, 1, 1
MIOpen(HIP): }
MIOpen(HIP): miopenStatus_t miopenDestroyTensorDescriptor(miopenTensorDescriptor_t){
MIOpen(HIP): 	tensorDesc = 4, 256, 336, 200
MIOpen(HIP): }
MIOpen(HIP): miopenStatus_t miopenDestroyTensorDescriptor(miopenTensorDescriptor_t){
MIOpen(HIP): 	tensorDesc = 4, 64, 336, 200
MIOpen(HIP): }
[4m[5m[31mERROR[0m [32m[06/22 19:13:16 ps.engine.train_loop]: [0mException during training:
Traceback (most recent call last):
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses/engine/train_loop.py", line 132, in train
    self.run_step()
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses/engine/train_loop.py", line 215, in run_step
    loss_dict = self.model(data)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses/modeling/meta_arch/rcnn.py", line 144, in forward
    features = self.backbone(images.tensor)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses/modeling/backbone/fpn.py", line 124, in forward
    bottom_up_features = self.bottom_up(x)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses/modeling/backbone/resnet.py", line 421, in forward
    x = stage(x)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/container.py", line 100, in forward
    input = module(input)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses/modeling/backbone/resnet.py", line 195, in forward
    out = self.conv3(out)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses/layers/wrappers.py", line 94, in forward
    x = super().forward(x)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/conv.py", line 349, in forward
    return self._conv_forward(input, self.weight)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/conv.py", line 346, in _conv_forward
    self.padding, self.dilation, self.groups)
RuntimeError: miopenStatusUnknownError
[32m[06/22 19:13:16 ps.engine.hooks]: [0mTotal training time: 0:00:32 (0:00:00 on hooks)
Traceback (most recent call last):
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/tools/train_net.py", line 224, in <module>
    args=(args,),
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses/engine/launch.py", line 57, in launch
    main_func(*args)
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/tools/train_net.py", line 185, in main
    return trainer.train()
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses/engine/defaults.py", line 422, in train
    super().train(self.start_iter, self.max_iter)
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses/engine/train_loop.py", line 132, in train
    self.run_step()
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses/engine/train_loop.py", line 215, in run_step
    loss_dict = self.model(data)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses/modeling/meta_arch/rcnn.py", line 144, in forward
    features = self.backbone(images.tensor)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses/modeling/backbone/fpn.py", line 124, in forward
    bottom_up_features = self.bottom_up(x)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses/modeling/backbone/resnet.py", line 421, in forward
    x = stage(x)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/container.py", line 100, in forward
    input = module(input)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses/modeling/backbone/resnet.py", line 195, in forward
    out = self.conv3(out)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "/public/home/ssct050t/users/zhangsy/projects/dets/pluses/pluses/layers/wrappers.py", line 94, in forward
    x = super().forward(x)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/conv.py", line 349, in forward
    return self._conv_forward(input, self.weight)
  File "/public/software/apps/DeepLearning/PyTorch/rocm3.3_torch1.5/lib/python3.6/site-packages/torch/nn/modules/conv.py", line 346, in _conv_forward
    self.padding, self.dilation, self.groups)
RuntimeError: miopenStatusUnknownError

```

**Comment by sunway513** [comment]
Hi @tonysy , it seems you have been using the ROCm3.3 based PyTorch. Can you try Detectron2 with ROCm3.5 based Pytorch build?
`docker pull rocm/pytorch:latest`
Plus, in the ROCm3.5 based docker container, you might want to enable the MIOpen hybrid find feature using the following environment variable:
`export MIOPEN_FIND_MODE=3`

**Comment by tonysy** [comment]
@sunway513 Thanks for your reply. I use an HPC cluster where I have no root access. I wonder is there possible to install ROCm3.5 and build the pytorch by myself without root access requirements.

If you can share a solution for this, it would be great.

**Comment by tonysy** [comment]
Or is there any possibility to solve the problem by setting some environment variables within ROCm3.3 environment. Thanks

**Comment by sunway513** [comment]
@tonysy there're two improvements in MIOpen from ROCm3.5 would help on Detectron2:
1. pre-compiled MIOpen kernels
2. Fast/Hybrid Find mode
However, none of those two were included in ROCm3.3. 
Can you create a github issue on the rocm/pytorch repo? We can discuss further there on the deployment options for your use case. 

https://github.com/ROCmSoftwarePlatform/pytorch

**Comment by tonysy** [comment]
@sunway513  Hi, I have created an issue and looking forward to the further discussion. Thanks a lot.
